File: Shared/FormsUtilities.vb
'*********************************************************************************************
' ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************
Imports System.Globalization
Imports System.Windows.Forms

Module FormsUtilities

    Function ErrorMessageBox(text As [String]) As DialogResult
        Return MessageBox.Show(text, Application.ProductName, MessageBoxButtons.OK, MessageBoxIcon.[Error], MessageBoxDefaultButton.Button1, DirectCast(0, MessageBoxOptions))
    End Function

    Function ErrorMessageBox(text As [String], ParamArray args As Object()) As DialogResult
        Return ErrorMessageBox([String].Format(CultureInfo.InvariantCulture, text, args))
    End Function

    Function ApplicationMessageBox(text As [String], buttons As MessageBoxButtons) As DialogResult
        Return MessageBox.Show(text, Application.ProductName, buttons, MessageBoxIcon.None, MessageBoxDefaultButton.Button1, DirectCast(0, MessageBoxOptions))
    End Function

    Function ApplicationMessageBox(text As [String], buttons As MessageBoxButtons, ParamArray args As Object()) As DialogResult
        Return ApplicationMessageBox([String].Format(CultureInfo.InvariantCulture, text, args), buttons)
    End Function

    Function InformationMessageBox(text As [String]) As DialogResult
        Return ApplicationMessageBox(text, MessageBoxButtons.OK)
    End Function

    Function InformationMessageBox(text As [String], ParamArray args As Object()) As DialogResult
        Return ApplicationMessageBox(text, MessageBoxButtons.OK, args)
    End Function

End Module

File: Shared/FormsUtilities.cs
// ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System.Globalization;
using System.Windows.Forms;

namespace SyncroSim.STSim
{
    internal static class FormsUtilities
    {
        public static DialogResult ErrorMessageBox(string text)
        {
            return MessageBox.Show(text, Application.ProductName, MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1, (MessageBoxOptions)0);
        }

        public static DialogResult ErrorMessageBox(string text, params object[] args)
        {
            return ErrorMessageBox(string.Format(CultureInfo.InvariantCulture, text, args));
        }

        public static DialogResult ApplicationMessageBox(string text, MessageBoxButtons buttons)
        {
            return MessageBox.Show(text, Application.ProductName, buttons, MessageBoxIcon.None, MessageBoxDefaultButton.Button1, (MessageBoxOptions)0);
        }

        public static DialogResult ApplicationMessageBox(string text, MessageBoxButtons buttons, params object[] args)
        {
            return ApplicationMessageBox(string.Format(CultureInfo.InvariantCulture, text, args), buttons);
        }

        public static DialogResult InformationMessageBox(string text)
        {
            return ApplicationMessageBox(text, MessageBoxButtons.OK);
        }

        public static DialogResult InformationMessageBox(string text, params object[] args)
        {
            return ApplicationMessageBox(text, MessageBoxButtons.OK, args);
        }
    }
}
